src/odemis/driver/zeiss.py

```diff
@@ -438,28 +469,37 @@ class Stage(model.Actuator):
     def __init__(self, name, role, parent, rng=None, **kwargs):
         """
         inverted (set of str): names of the axes which are inverted
-        rng (dict str -> (float,float)): axis name -> min/max of the position on
-          this axis. Note: if the axis is inverted, the range passed will be
-          inverted. Also, if the hardware reports position outside of the range,
-          move might fail, as it is considered outside of the range.
+        rng (dict str -> (float,float)): axis name -> min/max of the position on this axis. If an axis range is set to
+        None (null in YAML), it will be excluded and not controlled by Odemis.
+        By default, only x,y,z axes are controlled.
+        Note: if the axis is inverted, the range passed will be inverted.
+        Also, if the hardware reports position outside the range, move might fail, as it is considered outside the range.
         """
 
         if rng is None:
             rng = {}
 
+        # Ranges are from the documentation
         if "x" not in rng:
             rng["x"] = (5e-3, 152e-3)
         if "y" not in rng:
             rng["y"] = (5e-3, 152e-3)
         if "z" not in rng:
             rng["z"] = (5e-3, 40e-3)
-
-        axes_def = {
-            # Ranges are from the documentation
-            "x": model.Axis(unit="m", range=(rng["x"][0], rng["x"][1])),
-            "y": model.Axis(unit="m", range=(rng["y"][0], rng["y"][1])),
-            "z": model.Axis(unit="m", range=(rng["z"][0], rng["z"][1])),
-        }
+        # To keep compatibility with old configuration files, by default the extra axes are not used.
+        if "rx" not in rng:
+            rng["rx"] = None
+        if "rz" not in rng:
+            rng["rz"] = None
+        if "m" not in rng:  # Extra Z axis, in the Rx referential, to adjust eucentric height.
+            rng["m"] = None
+
+        axes_def = {}
+        for ax, r in rng.items():
+            if r:  # if range is not specified exclude the axis
```

Comment:
Please mark the comparison to `None` explicitly. It's not required by Python, but that makes things clearer when reading the code, and avoid also supporting `0`, `""` and `[]`.
```suggestion
            if r is not None:  # if range is not specified exclude the axis
```